### PR TITLE
Fix crash with missile overlays on no flagship

### PR DESCRIPTION
**Pull request by warp-core**
**Bugfix:**

Thanks to @Zitchas for reporting and helping debug this on Discord.

## Fix Details
The AlertLabel constructor dereferences the flagship pointer without checking that it's not null which causes a crash when attempting to draw missile overlays when there is no flagship, for example after captured in a failed boarding operation.
Curiously, it doesn't seem to happen when the flagship is destroyed in combat.

This change means all missile overlay labels will have the "hostile" colour if there is no flagship, since the other two colour options don't make sense if the flagship is gone.

We could alternatively just not draw the overlays at all if there is no flagship.

## Testing Done
Enable missile overlays in the settings and lose the flagship in a boarding operation while enemy missiles are in flight and see the game crash.

## Save File
This save file can be used to verify the bugfix. The bug will occur when using {{insert commit hash / version}}, and will not occur when using this branch's build.
[Rebbia Rampart~boardingcrashtest.txt](https://github.com/endless-sky/endless-sky/files/10420539/Rebbia.Rampart.boardingcrashtest.txt)
Thanks to @Zitchas for providing this save file.

### CODE REVIEW

File: source/AlertLabel.cpp [hunk 1]
```diff
@@ -36,10 +36,15 @@ AlertLabel::AlertLabel(const Point &position, const Projectile &projectile, cons
 		double zoom)
 	: position(position), zoom(zoom)
 {
-	isTargetingFlagship = projectile.TargetPtr() == flagship;
-	double maxHP = flagship->Attributes().Get("hull") + flagship->Attributes().Get("shield");
-	double missileDamage = projectile.GetWeapon().HullDamage() + projectile.GetWeapon().ShieldDamage();
-	bool isDangerous = (missileDamage / maxHP) > DANGEROUS_ABOVE;
+	bool isDangerous = false;
+	isTargetingFlagship = false;
+	if (flagship)
```
Comment (warp-core) [inline source/AlertLabel.cpp:41]:
```suggestion
	if(flagship)
```

**Review by Hurleveur — APPROVED** [review]
good fix

**Review by quyykk — APPROVED** [review]
Why not skip creating alert labels in the

**Comment by warp-core** [comment]
No thoughts on just not drawing them at all?

**Comment by quyykk** [comment]
I did lmao see the edit history of the approval lol. But there are other alerts too, and we should probably still show them?

**Comment by Zitchas** [comment]
From a player perspective, if they're continuing to watch the battle after being blown up, presumably they are interested in the course of the battle or watching how their allies do, or something like that. Would it be possible to have the overlay continue to only highlight missiles from hostile ships as threats. (basically, highlight as if the player was still there even though they aren't)

Or would that add a lot of complexity? (it's probably not worth much complexity, but it would be neat.

**Comment by warp-core** [comment]
These labels are only drawn for projectiles with non-zero missile strength from hostile governments.

```c++
	// Create missile overlays.
	missileLabels.clear();
	if(Preferences::Has("Show missile overlays"))
		for(const Projectile &projectile : projectiles)
		{
			Point pos = projectile.Position() - center;
			if(projectile.MissileStrength() && projectile.GetGovernment()->IsEnemy()
					&& (pos.Length() < max(Screen::Width(), Screen::Height()) * .5 / zoom))
				missileLabels.emplace_back(AlertLabel(pos, projectile, flagship, zoom));
		}
```

Not drawing them would mean changing it to this:

```c++
	// Create missile overlays.
	missileLabels.clear();
	if(Preferences::Has("Show missile overlays") && flagship)
		for(const Projectile &projectile : projectiles)
		{
			Point pos = projectile.Position() - center;
			if(projectile.MissileStrength() && projectile.GetGovernment()->IsEnemy()
					&& (pos.Length() < max(Screen::Width(), Screen::Height()) * .5 / zoom))
				missileLabels.emplace_back(AlertLabel(pos, projectile, flagship, zoom));
		}
```

**Comment by Hurleveur** [comment]
We could also just merge this hotfix and focus on the rest later

**Pull request merged**